core/trino-web-ui/src/main/resources/webapp-preview/src/App.tsx

```diff
@@ -11,41 +11,80 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import trinoLogo from './assets/trino.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { HashRouter as Router, Routes, Route, Navigate } from 'react-router-dom';
+import {
+  Button,
+  Container,
+  CssBaseline,
+  Unstable_Grid2 as Grid,
+  Link,
+  ThemeProvider,
+  Typography,
+  useMediaQuery
+} from '@mui/material';
+import { RootLayout as Layout } from './components/Layout';
+import { SnackbarProvider } from './components/SnackbarProvider';
+import { routers } from "./router.tsx";
+import { useConfigStore, Theme as ThemeStore } from './store';
+import { darkTheme, lightTheme } from './theme';
 
-function App() {
-  const [count, setCount] = useState(0)
+const App = () => {
+  const config = useConfigStore();
+  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
+
+  const themeToUse = () => {
+    if (config.theme === ThemeStore.Auto) {
+      return prefersDarkMode ? darkTheme : lightTheme;
+    } else if (config.theme === ThemeStore.Dark) {
+      return darkTheme;
+    } else {
+      return lightTheme;
+    }
+  }
 
   return (
     <>
-      <div>
-        <a href="https://vitejs.dev" target="_blank">
-          <img src={viteLogo} className="logo" alt="Vite logo"/>
-        </a>
-        <a href="https://react.dev" target="_blank">
-          <img src={reactLogo} className="logo react" alt="React logo"/>
-        </a>
-        <a href="https://trino.io" target="_blank">
-          <img src={trinoLogo} className="logo" alt="Trino logo"/>
-        </a>
-      </div>
-      <h1>Vite + React + Trino</h1>
-      <div className="card">
-        <button onClick={() => setCount((count) => count + 1)}>
-          count is {count}
-        </button>
-        <p>
-          Edit <code>src/App.tsx</code> and save to test HMR
-        </p>
-      </div>
-        <p>
-          <b>This is a placeholder of a potential new Trino Web UI</b>
-        </p>
+      <CssBaseline />
+      <ThemeProvider theme={themeToUse()}>
+        <SnackbarProvider>
+          <Router>
+            <Screen />
+          </Router>
+        </SnackbarProvider>
+      </ThemeProvider>
     </>
+  );
+}
+
+const Screen = () => {
+  return (
+    <Layout>
+      <Routes>
+        {routers.flatMap(router => {
+          return [<Route {...router.routeProps} key={router.itemKey}/>];
+        })}
+        <Route path="/" element={<Navigate to="/dashboard" />} />
+        <Route path="*" element={<NotFound />} key={"*"} />
+      </Routes>
+    </Layout>
+  );
+}
+
+const NotFound = () => {
```

Comment:
It'd be fun to include a Commander Bun Bun image as part of this.

Reply:
voila

![Screenshot 2024-08-05 220239](https://github.com/user-attachments/assets/f7a3e3d9-bf11-4304-a937-0dd9fe81368f)